cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -594,12 +606,18 @@ func (s *service) Stats(ctx context.Context, r *taskAPI.StatsRequest) (*taskAPI.
 	var statsx interface{}
 	switch cg := cgx.(type) {
 	case cgroup1.Cgroup:
+		if cg == nil {
+			return nil, errdefs.ToGRPCf(errdefs.ErrNotFound, "cgroup does not exist")
+		}
```

Comment:
This seems to be already checked here? https://github.com/containerd/containerd/blob/ed35ff3bfa38fcdf23c17e8ece8528e020b5514d/cmd/containerd-shim-runc-v2/task/service.go#L603-L605

Reply:
Thing is that `cgx` may not be `nil`, while its [underlying value](https://go.dev/tour/methods/12) `cg.(cgroup1.Cgroup)` can be `nil`. It's the same pattern that led to the nil pointer dereference in https://github.com/containerd/containerd/blob/ed35ff3bfa38fcdf23c17e8ece8528e020b5514d/cmd/containerd-shim-runc-v2/task/service.go#L280-L296 before the addition of the nil checks in the match arms. I added the logs of the panic I observed to the PR description.